src/weathergen/datasets/data_reader_synop.py

```diff
@@ -0,0 +1,274 @@
+# (C) Copyright 2025 WeatherGenerator contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+import logging
+from pathlib import Path
+from typing import override
+
+import numpy as np
+import xarray as xr
+from numpy.typing import NDArray
+
+from weathergen.datasets.data_reader_base import (
+    DataReaderTimestep,
+    ReaderData,
+    TimeWindowHandler,
+    TIndex,
+    check_reader_data,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class DataReaderSynop(DataReaderTimestep):
+    "Wrapper for SYNOP datasets from MetNo in NetCDF"
```

Comment:
Would it be useful to specify the requirements of the NetCDF file? Here is a suggestion:

```
Generic parser for station data in NetCDF format. The file must have 2 dimensions: time and location.
- Data variables must have dimensions in the following order (time, location). The names of the dimension can be anything.
- Geoinfo variables must have dimension (location,)
- Any variable with missing values must have the _FillValue attribute set
- A latitude and longitude variable with dimension (location,) must be provided. The units must be degrees and the variable name can be configured.
- A variable called time must be provided and have dimension (time,). Units must follow CF-conventions and the variable must have a units attribute.
```